libecho/tests/packet_tests.cpp

```diff
@@ -0,0 +1,266 @@
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#ifdef _WIN32
+#include <winsock2.h>
+#elif defined __unix__
+#include <arpa/inet.h>
+#endif
+
+#include "CRC.h"
+#include "Packet.h"
+
+uint16_t SIZE = 3, NUM = 2137;
+const std::vector<Flag> SET_FLAGS{Flag::ACK, Flag::DMD, Flag::LPC}, UNSET_FLAGS{Flag::SYN, Flag::FIN, Flag::RST};
+const std::vector<uint8_t> header{0, 38, 0, 3, 8, 89}, data{13, 4, 55}, incorrect_header{10, 37, 0, 4, 1, 2, 70},
+    incorrect_data_small{1}, incorrect_data_big{12, 34, 56, 78}, incorrect_crc_format_small{12},
+    incorrect_crc_format_big{12, 13, 14, 16, 17}, incorrect_crc{1, 1, 1, 1};
+
+template <typename T>
+union converter {
+    T a;
+    uint8_t b[sizeof(T)];
+};
+
+std::vector<uint8_t> calc_crc32(std::vector<uint8_t> v) {
+    converter<uint32_t> conv;
+    conv.a = htonl(CRC::Calculate(v.data(), v.size(), CRC::CRC_32()));
+    return std::vector<uint8_t>(conv.b, conv.b + 4);
+}
+
+TEST(tests_packet, test_loadHeaderFromBytes) {
+    auto p = Packet::loadHeaderFromBytes(header);
+    ASSERT_EQ(SIZE, p.getSize());
+    ASSERT_EQ(NUM, p.getNumber());
+    for (auto f : SET_FLAGS)
+        ASSERT_TRUE(p.isSet(f));
+    for (auto f : UNSET_FLAGS)
+        ASSERT_FALSE(p.isSet(f));
```

Comment:
Inconsistent style - later you use braces aroung single-line for, use them here too